src/Microsoft.AspNet.SignalR.Redis/RedisMessageBus.cs

```diff
@@ -27,6 +27,7 @@ public class RedisMessageBus : ScaleoutMessageBus
         private string _connectionString;
         private int _state;
         private readonly object _callbackLock = new object();
+        private readonly SemaphoreSlim _redisConnectionEventLock = new SemaphoreSlim(1, 1);
```

Comment:
because https://github.com/SignalR/SignalR/pull/3849/files/45b4089f1d23dc101284a7d7b945f246384baadc#diff-193a338ff4a86e8ec19aa6cfa2954768R189 (await expressions cannot be used in a synchronous function)